designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/NavigationBar.kt

```diff
@@ -0,0 +1,110 @@
+import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
+
+@Composable
+fun NavigationBar(
+    modifier: Modifier = Modifier,
+    items: List<NavigationBarItem>,
+    selectedItemIndex: Int? = null,
+    onItemClick: (Int) -> Unit,
+) {
+    Column(
+        modifier = modifier.testTag("NAVIGATION_BAR_CONTAINER"),
+    ) {
+        HorizontalDivider(
+            modifier = Modifier.testTag("NAVIGATION_BAR_BORDER"),
+            thickness = 0.6.dp,
+            color = Outline.Light,
+        )
+        androidx.compose.material3.NavigationBar(
+            modifier = Modifier.testTag("NAVIGATION_BAR"),
+            containerColor = SurfaceColor.ContainerLowest,
+        ) {
+            items.forEachIndexed { index, item ->
+                val selected = selectedItemIndex == index
+                NavigationBarItem(
+                    modifier = Modifier.testTag("NAVIGATION_BAR_ITEM_${item.label}"),
+                    colors = navigationBarItemColors(),
+                    icon = {
+                        NavigationBarItemIcon(item = item, selected = selected)
+                    },
+                    label = {
+                        Text(
+                            modifier = Modifier.testTag("NAVIGATION_BAR_ITEM_LABEL_${item.label}"),
+                            text = item.label,
+                        )
+                    },
+                    enabled = item.enabled,
+                    selected = selected,
+                    onClick = {
+                        onItemClick(index)
+                    },
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun NavigationBarItemIcon(item: NavigationBarItem, selected: Boolean) {
+    Box {
+        if (selected) {
+            item.selectedIcon()
+        } else {
+            item.defaultIcon()
+        }
+        if (item.showBadge) {
+            val badgeXOffset = if (!item.badgeText.isNullOrEmpty()) {
+                4.dp * item.badgeText.length
+            } else {
+                0.dp
+            }
+            val badgeYOffset = if (!item.badgeText.isNullOrEmpty()) (-4).dp else 0.dp
+
+            ErrorBadge(
+                modifier = Modifier.testTag("NAVIGATION_BAR_ITEM_BADGE_${item.label}")
+                    .align(Alignment.TopEnd)
+                    .offset(
+                        x = badgeXOffset,
+                        y = badgeYOffset,
+                    ),
+                text = item.badgeText,
+            )
+        }
+    }
+}
+
+@Composable
+fun navigationBarItemColors() = NavigationBarItemColors(
+    selectedIconColor = SurfaceColor.Primary,
+    selectedTextColor = TextColor.OnSurfaceVariant,
+    selectedIndicatorColor = SurfaceColor.Container,
+    unselectedIconColor = TextColor.OnSurfaceVariant,
+    unselectedTextColor = TextColor.OnSurfaceVariant,
+    disabledIconColor = TextColor.OnDisabledSurface,
+    disabledTextColor = TextColor.OnDisabledSurface,
+)
+
+data class NavigationBarItem(
```

Comment:
could you extract this class to a specific file?

Reply:
Done